osm2gtfs/creators/trips_creator.py

```diff
@@ -88,7 +88,8 @@ def _prepare_trips(self, feed, schedule, itinerary):
             if input_fr == itinerary.fr and input_to == itinerary.to:
                 trip_services = trip["services"]
                 for service in trip_services:
-                    services.append(service)
+                    if service not in services:
```

Comment:
To prevent the case when the `itineraries` in the `Schedule` are split by some reason, do not double the `services`.

Reply:
They are in some cases of the [Nicaraguan `timetable.json`](/mapanica/pt-data-nicaragua/blob/master/timetable.json) because of the generation by [easy-timetable-generator](/mapanica/easy-timetable-generator). But anyway, why shouldn't it be supported?

Reply:
I still don't understand it. Can you explain concretely why they should be split. Sorry for asking this stupid question, I just didn't get it, yet.

Reply:
I'm referring to the case when there are two or more dict items inside the line's list for the same line, which present different times but using the same service period. See this example:

```json
    "lines": {
        "E-MAN-EST": [
            {
                "exeptions": [],
                "from": "Mercado Mayoreo",
                "services": [
                    "Mo-Su"
                ],
                "stations": [
                    "Mercado Mayoreo",
                    "COTRAN Sur"
                ],
                "times": [
                    [
                        "05:45",
                        "08:15"
                    ],
                    [
                        "08:15",
                        "10:45"
                    ],
                    [
                        "09:15",
                        "11:45"
                    ],
                    [
                        "10:45",
                        "13:15"
                    ],
                    [
                        "11:45",
                        "14:15"
                    ]
                ],
                "to": "COTRAN Sur"
            },
            {
                "exeptions": [],
                "from": "Mercado Mayoreo",
                "services": [
                    "Mo-Su"
                ],
                "stations": [
                    "Mercado Mayoreo",
                    "COTRAN Sur"
                ],
                "times": [
                    [
                        "13:15",
                        "15:45"
                    ],
                    [
                        "15:15",
                        "17:45"
                    ],
                    [
                        "13:45",
                        "16:15"
                    ],
                    [
                        "15:45",
                        "18:15"
                    ]
                ],
                "to": "COTRAN Sur"
            },
            ...
```

Reply:
Thanks for the explanation. I think this (edge) case is kind of an inconsistency in the standard schedule format and I personally would rather require a clean input format. But I guess if you and @Skippern are in favour of supporting it, there is a deeper reason I still can't see.

Reply:
But shouldn't then the two dicts not been merged? As far as I see, now the second one (and all that follow) would be just discarded.

Reply:
> now the second one (and all that follow) would be just discarded

To clear it up a bit, I'll explain what happened before and what should happen IMHO:
* Before this PR, the `services` variable contained the following after iterating over the two (or more) dicts for one line: `[Mo-Su, Mo-Su]`. So in line [98](https://github.com/mapanica/osm2gtfs/blob/32524f075858f746dad0353620c900d51beb56ad/osm2gtfs/creators/trips_creator.py#L98) and more specifically in line 112, it would append the times from both dicts __twice__ to the `trips` variable with the same service period `Mo-Su`.
* With this change, the `service` variable would only contain the following: `[Mo-Su]` and the trips would be getting doubled.

This code isn't about the whole dict that would be discarded, but only preventing the doubling of a service period.

Reply:
Now I got it. Thanks!